tests/api/crud/test_secrets.py

```diff
@@ -0,0 +1,29 @@
+import fastapi.testclient
+import pytest
+import sqlalchemy.orm
+
+import mlrun.api.crud
+import mlrun.api.schemas
+import mlrun.errors
+
+
+def test_store_secrets_verifications(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient
+):
+    project = "project-name"
+    provider = mlrun.api.schemas.SecretProviderName.kubernetes
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={"invalid/key": "value"}
+            ),
+        )
+
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={"mlrun.internal.key": "value"}
+            ),
+        )
```

Comment:
Why not test success (with `allow_internal_secrets=True`)? Will require mocking the secrets functionality of course.

Reply:
Added full test for the internal secrets logic